Fix HTTPS support, allow self-signed certificates

letalumil opened the pull request:
Hey there!

First of all thank you for the fantastic library, it saved me tons of time!
In my particular use case I had to print over ipps, and I encountered with this error:
```javascript
Error: Protocol "https:" not supported. Expected "http:".
    at new ClientRequest (_http_client.js:52:11)
    at Object.exports.request (http.js:31:10)
    at module.exports (api/node_modules/ipp/lib/request.js:25:53)
    at Object.Printer.execute (api/node_modules/ipp/lib/printer.js:51:3)
    ...
```
The cause of the problem was a typo in [lib/request.js, line 25](https://github.com/williamkapke/ipp/blob/master/lib/request.js#L25). Missing colon after `https`.

I've fixed it, but also encountered with self-signed certificate problem, because my printer has it, so I had to add some flags to omit it. I guess it's [widely used approach](https://support.microsoft.com/en-us/kb/2021626). But may be it makes sense to make these flags optional for the sake of security?

Thanks,
Viktor

williamkapke commented on lib/request.js:
```diff
@@ -22,7 +22,19 @@ module.exports = function(opts, buffer, cb){
+		// Allow self-signet certs
+		opts.rejectUnauthorized = false;
+		opts.requestCert = true;
+		opts.agent = false;
```
I agree- these should be optional. They can be passed in.

letalumil commented:
I've removed hardcoded flags, bumped major version accordingly to the semver. I also fixed 'no-value' tag in the parses, let it be part of the PR?

For those, who interested how they can specify request flags:

```javascript
var parse = require('url').parse;

var url = parse("https://192.168.1.17:631/printers/HP-LaserJet-1020");

url.rejectUnauthorized = false;
url.requestCert = true;
url.agent = false;

var printer = ipp.Printer(url);
```

williamkapke commented:
published :tada:
